docs/configuration.md

```diff
@@ -491,6 +495,27 @@ map.
 >are equivalent, `layerinfo` has been deprecated, in favor or
 >`blobdescriptor`.
 
+### foreignlayers
+
+Use the `foreignlayers` subsection to restrict the hosts that may appear
+in foreign layer URLs of pushed manifests. (By default, any well-formed
+host is allowed.)
+
+Set the `hostwhitelist` field to a list of domains (strings beginning
+with `.`) and hosts (strings not beginning with `.`). Pushing a manifest
+will then succeed only if all foreign layer URLs contain hosts that
+match a list entry. Matching is case-insensitive, and domains entries
+match any subdomain while host entries must match exactly.
+
+The following example allows `example.com` and `WWW.EXAMPLE.NET` but not
+`WWW.EXAMPLE.COM` or `www.example.net`.
```

Comment:
I'm trying to separate the whitelist-a-single-name operation from the whitelist-all-subdomains-of-a-name operation.